tests/test_date.py

```diff
@@ -710,13 +713,74 @@ def then_date_object_is_invalid(self):
 
 
 class TestTimestampParser(BaseTestCase):
+    def given_parser(self, **params):
+        self.parser = date.DateDataParser(**params)
+
+    def given_tzstr(self, tzstr):
+        # Save the existing value
+        self.old_tzstr = os.environ['TZ'] if 'TZ' in os.environ else None
+
+        # Overwrite the value, or remove it
+        if tzstr is not None:
+            os.environ['TZ'] = tzstr
+        elif 'TZ' in os.environ:
+            del os.environ['TZ']
+
+        # Call tzset
+        tzset()
+
+    def reset_tzstr(self):
+        # If we never set it with given_tzstr, don't bother resetting it
+        if not hasattr(self, 'old_tzstr'):
+            return
+
+        # Restore the old value, or remove it if null
+        if self.old_tzstr is not None:
+            os.environ['TZ'] = self.old_tzstr
+        elif 'TZ' in os.environ:
+            del os.environ['TZ']
+
+        # Remove the local attribute
+        del self.old_tzstr
+
+        # Restore the old timezone behavior
+        tzset()
 
     def test_timestamp_in_milliseconds(self):
         self.assertEqual(
             date.get_date_from_timestamp('1570308760263', None),
             datetime.fromtimestamp(1570308760).replace(microsecond=263000)
         )
 
+    @parameterized.expand(
+        product(
+            ['1570308760'],
+            ['EDT', 'EST', 'PDT', 'PST', 'UTC', 'local'],
+            [None, 'EDT', 'EST', 'PDT', 'PST', 'UTC'],
+            ['EST5EDT4', 'UTC0', 'PST8PDT7', None],
+        )
+    )
+    def test_timestamp_with_different_timestr(self, timestamp, timezone, to_timezone, tzstr):
+        settings = {
+            'RETURN_AS_TIMEZONE_AWARE': True,
+            'TIMEZONE': timezone,
+        }
+
+        # is TO_TIMEZONE supposed to be allowed to be False, or None ???
```

Comment:
It’s `None` by default, so that value should be allowed, but `False` would be bad user input, I believe: https://dateparser.readthedocs.io/en/latest/settings.html#timezone-related-configurations

In order to test both passing `None` specifically and not defining the setting at all, you could define `DONT_SET = object()` in this module, add it to the `to_timezone` parameter list above, and use it in the `if` here instead of `None`.

Reply:
I got validation errors when I tried both `None` and `False`.

Reply:
The code seems to indicate that `False` is the default value? https://github.com/scrapinghub/dateparser/blob/21c629f067a40c2becb19c7e88ec8170a8da3fcb/dateparser_data/settings.py#L15 But I don't know exactly how `dateparser_data.settings` is used.

Reply:
OK, given your results and https://github.com/scrapinghub/dateparser/blob/21c629f067a40c2becb19c7e88ec8170a8da3fcb/dateparser/conf.py#L146-L150 I guess your current test code covers all it needs to cover.